### PR TITLE
IE: resolve the matchMedia problem

### LINKED ISSUE #2577: Visibility issues with Ads on Internet Explorer 11
Cube ads that are being displayed inside an iframe on IE11 are getting a visibility: hidden inline style added to it, which hides the ads. Does anyone know a workaround on this? This is not occurring on other acceptable browsers.

Here are three sites that currently have this problem. Each one shows fine as AMP pages in latest Chrome, Safari, Firefox, & Edge, but if you open these pages in IE11, cube ads are invisible. Our company is having the same problem and we are triaging it now.

1. https://www.washingtonpost.com/amphtml/blogs/plum-line/wp/2016/03/11/get-ready-donald-trump-youre-about-to-run-into-a-buzz-saw/#
2. http://mobile.nytimes.com/2016/03/10/opinion/sanders-surprises.amp.html#
3. http://www.people.com/people/article/amp/0,,20993119,00.html#

Not completely sure if this this a bug or if IE11 was intentionally left out because of other issues?

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #2577.

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -224,6 +232,50 @@ export class Resources {
...
+   */
+  fixMediaIe_(win) {
+    if (!this.platform_.isIe() || this.matchMediaIeQuite_(win)) {
+      this.relayoutAll_ = true;
+      return;
+    }
+
+    // Poll until the expression resolves correctly, but only up to a point.
+    const endTime = timer.now() + 2000;
+    const interval = win.setInterval(() => {
```
Comment (jridgewell) [inline src/service/resources-impl.js]: There are a few other polling places throughout the code base. Might be time for a poll helper, with optional `maxWait`?

Comment (dvoytenko) [inline src/service/resources-impl.js]: Yes. Good idea for `Timer` class perhaps. Please file the bug for fixit.

Comment (jridgewell) [inline src/service/resources-impl.js]: https://github.com/ampproject/amphtml/issues/3114

**Comment by cramforce** [comment]
My eyes.

LGTM

**Pull request merged**